### DISCUSSION TITLE
Forgot enabling `Microsoft-Windows-Subsystem-Linux`?

### DISCUSSION THREAD

**Discussion by BlackHole1** [podman-desktop/podman-desktop#3037] [Q&A] [answered]
Currently, on Windows, it is necessary to enable both `Microsoft-Windows-Subsystem-Linux` and `VirtualMachinePlatform` simultaneously (see: https://github.com/containers/podman/blob/4dc2e086186e0a32931949fa856816b73de45d74/pkg/machine/wsl/machine.go#L745-L753). However, in `podman-desktop`, only `VirtualMachinePlatform` is being detected. Is there any specific reason for this?

Ref: https://github.com/containers/podman-desktop/blob/bc24434f637c5490c4b7c58dc335ec13278ba03a/extensions/podman/src/podman-install.ts#L406-L427

/cc @evidolob

**Comment by jeffmaury** [comment]
There is also a WSL2Check: https://github.com/containers/podman-desktop/blob/bc24434f637c5490c4b7c58dc335ec13278ba03a/extensions/podman/src/podman-install.ts#L429

**Reply by BlackHole1** [reply]
Hey @jonboulle 👋. I have looked at this piece of code, but it still doesn't answer my question. In the `WSL2Check` class, it only checks if the WSL version is supported and if it has administrator privileges. But this is unrelated to my question about why there is no support for enabling `Microsoft-Windows-Subsystem-Linux`.

**Reply by BlackHole1** [reply]
See:
1. https://learn.microsoft.com/en-us/windows/wsl/install-manual#step-1---enable-the-windows-subsystem-for-linux
2. https://learn.microsoft.com/en-us/windows/wsl/install-manual#step-3---enable-virtual-machine-feature

**Reply by jeffmaury** [reply]
I would guess that this call will return an error is WSL is not set but we should wait for @evidolob

**Reply by evidolob** [reply, marked as answer]
> But this is unrelated to my question about why there is no support for enabling Microsoft-Windows-Subsystem-Linux.

Because we don't do any OS modification during installation, we just perform checks. So we don't enable even `VirtualMachinePlatform`, we just check that that is enabled. 
`Microsoft-Windows-Subsystem-Linux` not all Windows has that as separate feature(you could install it from marketplace, for example), so for `wsl2` we check command and it's output.

>I would guess that this call will return an error is WSL is not set but we should wait for @evidolob

Yes that true. 

The problem with Windows, that not all user account has right to enable windows features, so during podman installation we just check system requirements, and inform user if something is missing. And it up to user to solve them.

**Reply by BlackHole1** [reply]
Thanks, @jeffmaury and @evidolob for your reply!
